modules/javafx.controls/src/main/java/javafx/scene/control/TextArea.java

```diff
@@ -81,6 +82,51 @@
  * @since JavaFX 2.0
  */
 public class TextArea extends TextInputControl {
+    /** Denotes the function that moves the caret to the document end. */
```

Comment:
At a glance, this looks like we are polluting this `Control` class with `Behavior` related code.
This is another reason why I did not like the indirection of Key -> FunctionTag -> function.
Ideally, there should be no change to `Controls` classes while making `Behavior` classes public.

Imagine how weird the Javadoc for this class will look as we list all these FunctionTags.

Reply:
If one reads these function tag names, then it is obvious that they refer to the Behavior.
Hence, according to me, the correct place for them is in Behavior classes.
The fact that they are defined here and initialized in corresponding Behavior classes corroborates this.

If a user wants to customize the Behavior, he/she should do it by providing a custom Behavior class explicitly.